tests/lib.rs

```diff
@@ -263,7 +263,10 @@ pub fn layer_forward_helper(layer_type: &str, idims: Dim4, odims: Dim4, loss: &s
                   let h_t = utils::constant(hdims, DType::F64, 0.5f32);
 
                   // run a forward pass
-                  let (activ, _) = layer.forward(params.clone(), &x.clone(), Some(h_t));
+                  let (activ, _) = match layer_type {
+                      "unitary" => layer.forward(params.clone(), &x.clone(), None),
```

Comment:
To do a proper gradient test I suggest removing this and modifying the line above, i.e. `h_t` to be complex as you expect it. Having a zero-state is not a valid gradient check for an RNN